myjsonrpc.pm

```diff
@@ -23,19 +23,34 @@ use Errno;
 use Mojo::JSON;    # booleans
 use Cpanel::JSON::XS ();
 
+our $DEBUG_JSON = $ENV{PERL_MYJSONRPC_DEBUG} || 0;
+
 sub send_json {
     my ($to_fd, $cmd) = @_;
 
     # allow regular expressions to be automatically converted into
     # strings, using the Regex::TO_JSON function as defined at the end
     # of this file.
+    # The resulting JSON should be in a single line, otherwise
+    # read_json won't work
     my $cjx = Cpanel::JSON::XS->new->convert_blessed();
+    $cjx->canonical if $DEBUG_JSON;    # sort keys
```

Comment:
Why not just always use canonical JSON?

Reply:
> Why not just always use canonical JSON?

I don't know. Why wasn't `canonical` used before?

Reply:
> And perhaps UTF-8 encoding to make this more future proof?

ok

Reply:
resolved and force-pushed

Reply:
Fwiw. I think most JSON code in openQA now just uses the `Mojo::JSON` functions and [their defaults](https://github.com/mojolicious/mojo/blob/8b5acbbebc2030a675fcf896e63ed28c74d12679/lib/Mojo/JSON.pm#L33-L37) now anyway (for everything that doesn't want pretty printing).